uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/AbsenceType.tpl

```diff
@@ -270,11 +274,58 @@
                       {literal}
                   }
 
+                function initCalculationUnitControls() {
+                  setEntitlementLabelText();
+                  var prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
+                  $('.absence-calculation-unit select').on('change', function(e) {
+                    var add_public_holiday_entitlement = $('.public-holiday-option input[type=radio]:checked').val();
+                    var hours_unit_value = getHoursUnitValue();
+                    if(this.value == hours_unit_value && add_public_holiday_entitlement == 1) {
+                      $('.absence-calculation-unit select').select2('val', prev_calculation_unit_val);
+                      showPublicHolidayEntitlementErrorAlert();
+                    }
+                    else{
+                      prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
```

Comment:
I don't understand why the else is necessary. Would you please give more details?

Reply:
The `prev_calculation_unit_val` variable needs to be updated anytime the `Calculate Leave ` select field changes, even if its an allowed/valid change. This is to ensure that when the change is not valid (e.g selecting Hours option when add public holiday entitlement is yes), The field can always be set back to what it was before the user made the change.